src/main/Java/SampleApp.java

```diff
@@ -0,0 +1,267 @@
+	 * @throws IOException for REST HTTP POST
+	 */
+	public String getToken() throws IOException {
+		System.out.println("\nRunning example 1 to get token for authentication:");
+		System.out.println("Making REST HTTP POST call to " + props.getProperty("targetHost") + props.getProperty("authenticate") + 
+				" using API-key=" + props.getProperty("apiKey").substring(0, 8) + "...");
+		String tokenResponse = Request.Post(props.getProperty("targetHost") + props.getProperty("authenticate"))
+		.bodyString("{ \"userName\": \"" + props.getProperty("username") + "\","
+				+ " \"password\": \"" + props.getProperty("password") + "\" }", ContentType.APPLICATION_JSON)
+		.version(HttpVersion.HTTP_1_1)
+		.addHeader("Content-Type", "application/json")
+		.addHeader("API-key", props.getProperty("apiKey"))
+		.execute().returnContent().asString();
+		JsonObject jobj = new Gson().fromJson(tokenResponse, JsonObject.class);
+		System.out.println("Retrieved the following JSON from /auth/authenticate: \n" + jobj.toString());
+		String token = jobj.get("token").toString();
+		System.out.println("Retrieved token " + token);
+		return token;
+	}
+	
+	/**
+	 * Uses HTTP GET to obtain the JSON response as a string from the /search?searchTerm= API call
+	 * @throws IOException for REST HTTP GET
+	 * @return String with the searchResult
+	 */
+	public String getSearchResult() throws IOException{
+		System.out.println("\nRunning example 2 to search for certified products:");
+		System.out.println("Making REST HTTP GET call to " + props.getProperty("targetHost") + props.getProperty("search") + 
+				" using API-key=" + props.getProperty("apiKey").substring(0, 8) + "...");
+		String searchResult = Request.Get(props.getProperty("targetHost") + props.getProperty("search"))
+				.version(HttpVersion.HTTP_1_1)
+				.addHeader("Content-Type", "application/json")
+				.addHeader("API-key", props.getProperty("apiKey"))
+				.execute().returnContent().asString();
+		System.out.println("Retrieved " + props.getProperty("targetHost") + props.getProperty("search") + " JSON as follows: \n" + searchResult);
+		return searchResult;
+	}
+
+	/**
+	 * Uses HTTP GET to obtain the JSON response from the /search?searchTerm= API call and save as a csv
+	 * @throws IOException for REST HTTP GET
+	 */
+	public void getSearchResult_saveResultsInFile() throws IOException{
+		System.out.println("\nRunning example 3 to search for certified products and save JSON in a text file:");
+		System.out.println("Making REST HTTP GET call to " + props.getProperty("targetHost") + props.getProperty("search") + 
+				" using API-key=" + props.getProperty("apiKey").substring(0, 8) + "..." + " and saving JSON to fullResults.txt");
+		Request.Get(props.getProperty("targetHost") + props.getProperty("search"))
+				.version(HttpVersion.HTTP_1_1)
+				.addHeader("Content-Type", "application/json")
+				.addHeader("API-key", props.getProperty("apiKey"))
+				.execute().saveContent(new File("fullResults.txt"));
```

Comment:
I didn't realize what fullResults.txt was (checked in above) but now I'm wondering if that should really be checked in filled up json data. Seems like that file should not exist in git and should only get created (and probably ignored) when someone chooses this menu option.